pkg/ingester/limiter.go

```diff
@@ -214,6 +214,28 @@ func (l *Limiter) maxMetadataPerUser(userID string) int {
 	)
 }
 
+func (l *Limiter) maxExemplarsPerUser(userID string) int {
+	// Return minimum nonzero limit, or 0 if no limits applied.
+	// Because the exemplar value cannot be changed at runtime, it supports
+	// only fixed values, either a local per-ingester limit, or
+	// global limited divided by fixed shuffle shard size when both shuffle sharding
+	// and shard by all labels are enabled.
+
+	limit := l.limits.MaxLocalExemplarsPerUser(userID)
+
+	if globalLimit := l.limits.MaxGlobalExemplarsPerUser(userID); globalLimit > 0 && l.shardByAllLabels {
```

Comment:
Could you elaborate why when `shardByAllLabels` is disabled we can't have the global limit?